index.js

```diff
@@ -67,18 +67,28 @@ server.get("/cadastro/:slug", (request, response) => {
 
   const html = jetpack.read(__dirname+"/views/editar.html");
 
-  const formattedHtml = Eta.render(html, { ...artigo, slug })
+  const formattedHtml = Eta.render(html, { ...artigo, slug})
 
   response.send(formattedHtml);
 });
 
 server.post("/cadastro", (request, response) => {
-  const artigo = request.body;
+  const formArticle = request.body;
 
   const database = JSON.parse(jetpack.read('./database.json'));
-  database.article[artigo.url] = {
-    title: artigo.titulo,
-    content: artigo.conteudo,
+  const databaseArticle = database.article[formArticle.url];
+
+  let date = "";
+  if (databaseArticle?.date === undefined) {
+    date = createDate();
+  } else {
+    date = databaseArticle.date;
+  }
+
+  database.article[formArticle.url] = {
+    title: formArticle.titulo,
+    content: formArticle.conteudo,
+    date,
   };
```

Comment:
Para evitar o IF acima, com o ajuste de retornar um objeto vazio em vez de `undefined`, seria seguro fazer isso
```suggestion
  database.article[formArticle.url] = {
    date: new Date().toJSON(),
     ...databaseArticle,
    title: formArticle.titulo,
    content: formArticle.conteudo
  };
```
Esse trecho `...databaseArticle` vai substituir todo substituir os valores já existentes em `databaseArticle` pelas keys acima, como `date` isso evita editar ela, mas tudo que está abaixo dele, pode substituir oque já existe, isso evita o seu IF que evitava editar a `date`.

Reply:
Corrigido.